<link rel="import" href="/bower_components/prism-js/prism-js.html">
<template><div class="wikiArticle text-content redesign">
<div>
  <div class="overheadIndicator nonStandard nonStandardHeader">
      <p><strong>Non-standard</strong><br>
      This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future.</p>
      </div></div>
<h2 id="Summary" name="Summary">Summary</h2>
<p>The <strong><code>InternalError</code> object</strong> indicates an error that occurred internally in the JavaScript engine. For example: <strong>&quot;InternalError</strong>: too much recursion&quot;.</p>
<h2 id="Syntax" name="Syntax">Syntax</h2>
<pre class="syntaxbox"><code>new InternalError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>
<h3 id="Parameters" name="Parameters">Parameters</h3>
<dl>
 <dt>
  <code>message</code></dt>
 <dd>
  Optional. Human-readable description of the error</dd>
 <dt>
  <code>fileName</code> <span title="This API has not been standardized."><i class="icon-warning-sign"> </i></span></dt>
 <dd>
  Optional. The name of the file containing the code that caused the exception</dd>
 <dt>
  <code>lineNumber</code> <span title="This API has not been standardized."><i class="icon-warning-sign"> </i></span></dt>
 <dd>
  Optional. The line number of the code that caused the exception</dd>
</dl>
<h2 id="Description" name="Description">Description</h2>
<p>An <code>InternalError</code> is thrown whenever an internal error in the JavaScript engine occurred.</p>
<p>Example cases are mostly when something is too large, e.g.:</p>
<ul>
 <li>&quot;too many switch cases&quot;,</li>
 <li>&quot;too many parentheses in regular expression&quot;,</li>
 <li>&quot;array initializer too large&quot;,</li>
 <li>&quot;too much recursion&quot;.</li>
</ul>
<h2 id="Properties" name="Properties">Properties</h2>
<dl>
 <dt>
  <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/InternalError/prototype" title="The InternalError.prototype property represents the prototype of the InternalError constructor."><code>InternalError.prototype</code></a></dt>
 <dd>
  Allows the addition of properties to an <code>InternalError</code> object.</dd>
</dl>
<h2 id="Methods" name="Methods">Methods</h2>
<p>The global <code>InternalError</code> contains no methods of its own, however, it does inherit some methods through the prototype chain.</p>
<h2 id="InternalError_instances" name="InternalError_instances"><code>InternalError</code> instances</h2>
<h3 id="Properties_of_InternalError_instances" name="Properties_of_InternalError_instances">Properties</h3>
<div>
 <dl><dt><code>InternalError.prototype.constructor</code></dt>
 <dd>Specifies the function that created an instance&apos;s prototype.</dd>
 <dt><a href="#/Error.prototype.message" title="The message property is a human-readable description of the error."><code>InternalError.prototype.message</code></a></dt>
 <dd>Error message. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
 <dt><a href="#/Error.prototype.name" title="The name property represents a name for the type of error. The initial value is &quot;Error&quot;."><code>InternalError.prototype.name</code></a></dt>
 <dd>Error name. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
 <dt><a href="#/Error.prototype.fileName" title="The fileName property contains the path to the file that raised this error."><code>InternalError.prototype.fileName</code></a></dt>
 <dd>Path to file that raised this error. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
 <dt><a href="#/Error.prototype.lineNumber" title="The lineNumber property contains the line number in the file that raised this error."><code>InternalError.prototype.lineNumber</code></a></dt>
 <dd>Line number in file that raised this error. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
 <dt><a href="#/Error.prototype.columnNumber" title="The columnNumber property contains the column number in the line of the file that raised this error."><code>InternalError.prototype.columnNumber</code></a></dt>
 <dd>Column number in line that raised this error. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
 <dt><a href="#/Error.prototype.stack" title="The non-standard stack property of Error objects offer a trace of which functions were called, in what order, from which line and file, and with what arguments. The stack string proceeds from the most recent calls to earlier ones, leading back to the original global scope call."><code>InternalError.prototype.stack</code></a></dt>
 <dd>Stack trace. Inherited from <a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a>.</dd>
</dl></div>
<h3 id="Methods_of_InternalError_instances" name="Methods_of_InternalError_instances">Methods</h3>
<div>
 <p>Although the <a href="#/InternalError" title="The InternalError object indicates an error that occured internally in the JavaScript engine. For example: &quot;InternalError: too much recursion&quot;."><code>InternalError</code></a> prototype object does not contain any methods of its own, <a href="#/InternalError" title="The InternalError object indicates an error that occured internally in the JavaScript engine. For example: &quot;InternalError: too much recursion&quot;."><code>InternalError</code></a> instances do inherit some methods through the prototype chain.</p></div>
<h2 id="Specifications">Specifications</h2>
<p>Not part of any specifications.</p>
<h2 id="See_also" name="See_also">See also</h2>
<ul>
 <li><a href="#/Error" title="The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base objects for user-defined exceptions. See below for standard built-in error types."><code>Error</code></a></li>
 <li><a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/InternalError/prototype" title="The InternalError.prototype property represents the prototype of the InternalError constructor."><code>InternalError.prototype</code></a></li>
</ul>
<div class="mdnLicenseFooter">This page on MDN: <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/InternalError">InternalError</a>, by Mozilla Contributors.<br>Licensed under the terms of Creative Commons Attribution-ShareAlike.</div>
</div></template>